src/app/src/components/annotations/annotator.tsx

```diff
@@ -552,39 +556,82 @@ export default class Annotator extends Component<
     this.setState({ annotatedAssetsHidden: flag });
   }
 
+  // eslint-disable-next-line react/sort-comp
+  private async bulkAnalysis() {
+    /* Blocker to account for case where there is no model to perform prediction */
+    if (!this.props.loadedModel) {
+      CreateGenericToast("There is no model loaded", Intent.WARNING, 3000);
+      return;
+    }
+    this.setState({ predictTotal: 100, predictDone: 0.01, multiplier: 1 });
+    this.setState({ uiState: "Predicting" });
+    this.handleProgressToast();
+
+    const lastAsset = this.state.assetList[this.state.assetList.length - 1];
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const asset of this.state.assetList) {
+      this.selectAsset(asset, false);
+      // eslint-disable-next-line no-await-in-loop
+      await this.getInference(asset, true, asset.url === lastAsset.url);
+    }
+    await this.updateImage();
+    this.setState({ predictDone: 0, uiState: null });
+  }
+
   /**
-   * Centralized Handler to Perform predictions on both Video and Images
+   * Perform single predictions fore either Video or Image
    */
-  private async getInference(reanalyse = true) {
+  private async singleAnalysis(reanalyse = true) {
     /* Blocker to account for case where prediction is still running */
     if (this.state.predictDone !== 0 || this.state.uiState === "Predicting") {
       CreateGenericToast("Inference is already running", Intent.WARNING, 3000);
       return;
     }
 
-    /* Blocker to account for case where there is no model to perform prediction */
     if (!this.props.loadedModel) {
       CreateGenericToast("There is no model loaded", Intent.WARNING, 3000);
       return;
     }
 
-    const loadedModelHash = this.props.loadedModel.hash;
-
     this.setState({ predictTotal: 100, predictDone: 0.01, multiplier: 1 });
     this.setState({ uiState: "Predicting" });
-    if (reanalyse) {
-      this.handleProgressToast();
+    this.handleProgressToast();
+    await this.getInference(this.currentAsset, reanalyse);
+    await this.updateImage();
```

Comment:
Looks good to me! Correct me if I'm wrong but I'm assuming that `await` will pause the execution of line 601 until 600 completes with response?

Reply:
yeap